### PR TITLE
[WIP] Add IApplicationBuilder extension method UseSignalRWithSpecification

### PR DESCRIPTION
**Pull request by RicoSuter**

### CODE REVIEW

File: src/HelloSignalR/Startup.cs [hunk 1]
```diff
@@ -14,10 +20,76 @@ public void ConfigureServices(IServiceCollection services)
         public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             app.UseStaticFiles();
-            app.UseSignalR(routes =>
+            app.UseSignalRWithSpecification(routes =>
```
Comment (bradygaster) [inline src/HelloSignalR/Startup.cs]: All due respect, i appreciate your consistency here but I am concerned mixing the actual SignalR wire-up and the specification/description in the same breath might confuse some. For instance, what if someone wants to wire up the Azure SignalR Service here? Would you add pass-throughs here to enable folks to do this, too?

Comment (RSuter) [reply src/HelloSignalR/Startup.cs]: Yes, i dont like this too, that’s why i’m asking for an api explorer... without that there are only two options: This interceptor/wrapper approach or specifing the hubs twice (one for the spec and one for signalr)... this PR is just a prototype/my sample impl

Comment (RicoSuter) [reply src/HelloSignalR/Startup.cs]: For core 3.0 this can now be implemented, see also

https://github.com/dotnet/aspnetcore/pull/8425

**Comment by RicoSuter** [comment]
For core 3.0 this can now be implemented correctly, see also

https://github.com/dotnet/aspnetcore/pull/8425

**Pull request closed without merging**